src/components/search.js

```diff
@@ -27,16 +40,35 @@ export default class Search extends React.PureComponent {
     )
   }
 
+  handleClick() {
+    if (this.input.value == '') return
+    this.clear()
+
+    this.props.onSearch(
+      this.emojiIndex.search('', {
+        emojisToShowFilter: this.props.emojisToShowFilter,
+        maxResults: this.props.maxResults,
+        include: this.props.include,
+        exclude: this.props.exclude,
+        custom: this.props.custom,
+      }),
+    )
+  }
+
   setRef(c) {
     this.input = c
   }
 
   clear() {
     this.input.value = ''
+    this.setState({
+      icon: SVGs.searching['search'],
+    })
```

Comment:
We've got the same `setState` above in `handleChange`. 

Let's add that to the `search` method I just suggested above:

```js
  search(value) {
    if (value == '') {
      this.setState({
        icon: SVGs.searching['search'],
      })
    } else {
      this.setState({
        icon: SVGs.searching['delete'],
      })
    }
    this.props.onSearch(
      this.emojiIndex.search(value, {
        emojisToShowFilter: this.props.emojisToShowFilter,
        maxResults: this.props.maxResults,
        include: this.props.include,
        exclude: this.props.exclude,
        custom: this.props.custom,
      }),
    )
  }

  handleChange() {
    this.search(this.input.value)
  }

  handleClick() {
    if (this.input.value == '') return
    this.input.value = ''
    this.search('')
  }
```

and we can delete `clear`?